drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -73,4 +80,57 @@ def factory_name
   def schema
     raise 'Implement in sub-class'
   end
+
+  def process_custom_field(field, value)
+    record = @processor.send(factory_name)
+    return false unless record.respond_to?(:custom_data_elements)
+
+    cded = Hmis::Hud::CustomDataElementDefinition.for_type(record.class.sti_name).find_by(key: field)
+    return false unless cded.present?
+
+    attrs = {
+      user: @processor.hud_user,
+      data_source_id: @processor.hud_user.data_source_id,
+      data_element_definition: cded,
+    }
+    value_field_name = "value_#{cded.field_type}"
+    value = attribute_value_for_enum(nil, value) # converts HIDDEN => nil
+
+    existing_values = record.custom_data_elements.where(data_element_definition: cded, owner: record)
+
+    # If this custom field only allows 1 value and there already is one, update it.
+    if !cded.repeats && existing_values.exists?
+      cde_attributes = { id: existing_values.first.id }
+      if value.present?
+        cde_attributes[value_field_name] = value
+        cde_attributes.merge!(attrs)
+      else
+        cde_attributes[:_destroy] = 1
+      end
+
+      record.assign_attributes(
+        custom_data_elements_attributes: [
+          # Update or delete the existing value
+          cde_attributes,
+          # Delete any other values (there shouldn't be any, but just in case)
```

Comment:
This is actually already validated on the `Hmis::Hud::CustomDataElement` as a `validates_uniqueness_of` validator. It could be done at the database level instead as a partial index - is that preferred to doing it as an AR validation?

Either way, we don't need this here.. I was just having a hard time trusting the validator ;) I'll remove it.